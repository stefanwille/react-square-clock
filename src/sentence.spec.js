const { mapDateToSentence, mapTimeToSentence } = require('./sentence');

describe('mapDateToSentence()', () => {
  it('takes a date and maps it to a display time', () => {
    const date = new Date('2017-12-16T21:10:09+01:00');
    expect(mapDateToSentence(date)).toEqual({ minutes: 'ZEHN', relation: 'NACH', hours: 'NEUN' });
  });
});

describe('mapTimeToSentence()', () => {
  it('maps a 12 hour time to a sentence', () => {
    expect(mapTimeToSentence('9:10')).toEqual({ minutes: 'ZEHN', relation: 'NACH', hours: 'NEUN' });
  });

  it('treats the minutes 1-10 as X NACH', () => {
    expect(mapTimeToSentence('9:01')).toEqual({ minutes: 'EINS', relation: 'NACH', hours: 'NEUN' });
    expect(mapTimeToSentence('9:02')).toEqual({ minutes: 'ZWEI', relation: 'NACH', hours: 'NEUN' });
    expect(mapTimeToSentence('9:10')).toEqual({ minutes: 'ZEHN', relation: 'NACH', hours: 'NEUN' });
  });

  it('treats the minutes 11-12 as ZEHN NACH', () => {
    expect(mapTimeToSentence('9:11')).toEqual({ minutes: 'ZEHN', relation: 'NACH', hours: 'NEUN' });
    expect(mapTimeToSentence('9:12')).toEqual({ minutes: 'ZEHN', relation: 'NACH', hours: 'NEUN' });
  });

  it('treats the minutes 13-17 as VIERTEL NACH', () => {
    expect(mapTimeToSentence('9:13')).toEqual({
      fraction: 'VIERTEL',
      relation: 'NACH',
      hours: 'NEUN',
    });
    expect(mapTimeToSentence('9:15')).toEqual({
      fraction: 'VIERTEL',
      relation: 'NACH',
      hours: 'NEUN',
    });
    expect(mapTimeToSentence('9:17')).toEqual({
      fraction: 'VIERTEL',
      relation: 'NACH',
      hours: 'NEUN',
    });
  });

  it('treats the minutes 18-35 as HALB', () => {
    expect(mapTimeToSentence('9:30')).toEqual({
      fraction: 'HALB',
      hours: 'ZEHN',
    });
    expect(mapTimeToSentence('9:18')).toEqual({
      fraction: 'HALB',
      hours: 'ZEHN',
    });
    expect(mapTimeToSentence('9:35')).toEqual({
      fraction: 'HALB',
      hours: 'ZEHN',
    });
  });

  it('treats the minutes 36-43 as ZWANZIG VOR', () => {
    expect(mapTimeToSentence('9:36')).toEqual({
      minutes: 'ZWANZIG',
      relation: 'VOR',
      hours: 'ZEHN',
    });
    expect(mapTimeToSentence('9:40')).toEqual({
      minutes: 'ZWANZIG',
      relation: 'VOR',
      hours: 'ZEHN',
    });
    expect(mapTimeToSentence('9:43')).toEqual({
      minutes: 'ZWANZIG',
      relation: 'VOR',
      hours: 'ZEHN',
    });
  });

  it('treats the minutes 44-47 as VIERTEL VOR', () => {
    expect(mapTimeToSentence('9:44')).toEqual({
      fraction: 'VIERTEL',
      relation: 'VOR',
      hours: 'ZEHN',
    });
    expect(mapTimeToSentence('9:45')).toEqual({
      fraction: 'VIERTEL',
      relation: 'VOR',
      hours: 'ZEHN',
    });
    expect(mapTimeToSentence('9:47')).toEqual({
      fraction: 'VIERTEL',
      relation: 'VOR',
      hours: 'ZEHN',
    });
  });

  it('treats the minutes 48-50 as ZEHN VOR', () => {
    expect(mapTimeToSentence('9:48')).toEqual({
      minutes: 'ZEHN',
      relation: 'VOR',
      hours: 'ZEHN',
    });
    expect(mapTimeToSentence('9:50')).toEqual({
      minutes: 'ZEHN',
      relation: 'VOR',
      hours: 'ZEHN',
    });
  });

  it('treats minutes 51-59 as X VOR', () => {
    expect(mapTimeToSentence('9:51')).toEqual({
      minutes: 'NEUN',
      relation: 'VOR',
      hours: 'ZEHN',
    });
    expect(mapTimeToSentence('9:55')).toEqual({
      minutes: 'FÃœNF',
      relation: 'VOR',
      hours: 'ZEHN',
    });
  });

  it('treats 9:00 as ES IST 9 UHR', () => {
    expect(mapTimeToSentence('9:00')).toEqual({
      it: 'ES',
      is: 'IST',
      hours: 'NEUN',
      oclock: 'UHR',
    });
  });
});
