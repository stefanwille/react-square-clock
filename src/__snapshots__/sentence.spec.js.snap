// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapTimeToSentence() all hours maps 01:05 1`] = `"FÜNF NACH EINS"`;

exports[`mapTimeToSentence() all hours maps 02:05 1`] = `"FÜNF NACH ZWEI"`;

exports[`mapTimeToSentence() all hours maps 03:05 1`] = `"FÜNF NACH DREI"`;

exports[`mapTimeToSentence() all hours maps 04:05 1`] = `"FÜNF NACH VIER"`;

exports[`mapTimeToSentence() all hours maps 05:05 1`] = `"FÜNF NACH FÜNF"`;

exports[`mapTimeToSentence() all hours maps 06:05 1`] = `"FÜNF NACH SECHS"`;

exports[`mapTimeToSentence() all hours maps 07:05 1`] = `"FÜNF NACH SIEBEN"`;

exports[`mapTimeToSentence() all hours maps 08:05 1`] = `"FÜNF NACH ACHT"`;

exports[`mapTimeToSentence() all hours maps 09:05 1`] = `"FÜNF NACH NEUN"`;

exports[`mapTimeToSentence() all hours maps 10:05 1`] = `"FÜNF NACH ZEHN"`;

exports[`mapTimeToSentence() all hours maps 11:05 1`] = `"FÜNF NACH ELF"`;

exports[`mapTimeToSentence() all hours maps 12:05 1`] = `"FÜNF NACH ZWÖLF"`;

exports[`mapTimeToSentence() all minutes maps 03:00 1`] = `"ES IST DREI UHR"`;

exports[`mapTimeToSentence() all minutes maps 03:01 1`] = `"ES IST DREI UHR"`;

exports[`mapTimeToSentence() all minutes maps 03:02 1`] = `"ES IST DREI UHR"`;

exports[`mapTimeToSentence() all minutes maps 03:03 1`] = `"DREI NACH DREI"`;

exports[`mapTimeToSentence() all minutes maps 03:04 1`] = `"VIER NACH DREI"`;

exports[`mapTimeToSentence() all minutes maps 03:05 1`] = `"FÜNF NACH DREI"`;

exports[`mapTimeToSentence() all minutes maps 03:06 1`] = `"FÜNF NACH DREI"`;

exports[`mapTimeToSentence() all minutes maps 03:07 1`] = `"FÜNF NACH DREI"`;

exports[`mapTimeToSentence() all minutes maps 03:08 1`] = `"ZEHN NACH DREI"`;

exports[`mapTimeToSentence() all minutes maps 03:09 1`] = `"ZEHN NACH DREI"`;

exports[`mapTimeToSentence() all minutes maps 03:10 1`] = `"ZEHN NACH DREI"`;

exports[`mapTimeToSentence() all minutes maps 03:11 1`] = `"ZEHN NACH DREI"`;

exports[`mapTimeToSentence() all minutes maps 03:12 1`] = `"ZEHN NACH DREI"`;

exports[`mapTimeToSentence() all minutes maps 03:13 1`] = `"VIERTEL NACH DREI"`;

exports[`mapTimeToSentence() all minutes maps 03:14 1`] = `"VIERTEL NACH DREI"`;

exports[`mapTimeToSentence() all minutes maps 03:15 1`] = `"VIERTEL NACH DREI"`;

exports[`mapTimeToSentence() all minutes maps 03:16 1`] = `"VIERTEL NACH DREI"`;

exports[`mapTimeToSentence() all minutes maps 03:17 1`] = `"VIERTEL NACH DREI"`;

exports[`mapTimeToSentence() all minutes maps 03:18 1`] = `"ZWANZIG NACH DREI"`;

exports[`mapTimeToSentence() all minutes maps 03:19 1`] = `"ZWANZIG NACH DREI"`;

exports[`mapTimeToSentence() all minutes maps 03:20 1`] = `"ZWANZIG NACH DREI"`;

exports[`mapTimeToSentence() all minutes maps 03:21 1`] = `"ZWANZIG NACH DREI"`;

exports[`mapTimeToSentence() all minutes maps 03:22 1`] = `"ZWANZIG NACH DREI"`;

exports[`mapTimeToSentence() all minutes maps 03:23 1`] = `"ZWANZIG NACH DREI"`;

exports[`mapTimeToSentence() all minutes maps 03:24 1`] = `"ZWANZIG NACH DREI"`;

exports[`mapTimeToSentence() all minutes maps 03:25 1`] = `"HALB VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:26 1`] = `"HALB VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:27 1`] = `"HALB VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:28 1`] = `"HALB VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:29 1`] = `"HALB VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:30 1`] = `"HALB VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:31 1`] = `"HALB VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:32 1`] = `"HALB VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:33 1`] = `"HALB VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:34 1`] = `"HALB VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:35 1`] = `"HALB VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:36 1`] = `"ZWANZIG VOR VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:37 1`] = `"ZWANZIG VOR VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:38 1`] = `"ZWANZIG VOR VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:39 1`] = `"ZWANZIG VOR VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:40 1`] = `"ZWANZIG VOR VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:41 1`] = `"ZWANZIG VOR VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:42 1`] = `"ZWANZIG VOR VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:43 1`] = `"VIERTEL VOR VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:44 1`] = `"VIERTEL VOR VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:45 1`] = `"VIERTEL VOR VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:46 1`] = `"VIERTEL VOR VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:47 1`] = `"VIERTEL VOR VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:48 1`] = `"ZEHN VOR VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:49 1`] = `"ZEHN VOR VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:50 1`] = `"ZEHN VOR VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:51 1`] = `"ZEHN VOR VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:52 1`] = `"ZEHN VOR VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:53 1`] = `"FÜNF VOR VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:54 1`] = `"FÜNF VOR VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:55 1`] = `"FÜNF VOR VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:56 1`] = `"VIER VOR VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:57 1`] = `"DREI VOR VIER"`;

exports[`mapTimeToSentence() all minutes maps 03:58 1`] = `"ES IST VIER UHR"`;

exports[`mapTimeToSentence() all minutes maps 03:59 1`] = `"ES IST VIER UHR"`;
